### DISCUSSION TITLE
"Reply to this email directly" includes email thread/history

### DISCUSSION THREAD

**Discussion by pzcfg** [community/community#77491] [Projects and Issues]
### Select Topic Area

Bug

### Feature Area

Issues

### Body

I tried using the "Reply this email directly" feature for an issue thread for the first time last night

![Reply via email](https://github.com/community/community/assets/94077014/a8dc9c35-b4d2-4aae-a39e-9fe2b9327cd2)

I decided to go to the website just to check if/how it went through, and was disappointed to discover that it included all of the usual "On [date], [email] wrote:" content of the reply. Worse still, despite there being new lines between my message and the reply content, GitHub seemingly deleted all of this, leading to the "somethingOn" at the end of my actual message

![Reply mess](https://github.com/community/community/assets/94077014/088537e7-34fa-43ac-bb04-ac3d8e7c42bc)

Not sure if this behavior is intentional or not, but most "reply via email" systems (and all of the ones I have used) have detected and deleted this reply history content.

---

Additionaly, when I went to fix this on the web, I discovered that I am not able to use/enable markdown for the edit, even though I'm now on the web.

<img width="930" alt="Can't markdown" src="https://github.com/community/community/assets/94077014/8c661493-49b2-48a5-a362-57380403e351">

Personally, I think markdown _should_ be allowed via email, but if that's not desired, I should at least be able to enable it if I go to edit from the web.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by zsloane** [comment]
This is even worse when you view the issue within a project. It strips out all newlines etc so it is just a jumbled mess. It is at least readable when you view the issue in whatever repo it lives in. 

Not to mention the fact that the history grows with every comment and ends up having redacted emails everywhere, multiple links to the issue (which makes no sense since it is now in a comment on the issue itself), "unsubscribe from notifications" links, and socials links or other poorly parsed junk from email signatures.

This just needs to be turned off completely

**Comment by tiagov8** [comment]
Added a related suggestion: https://bettersoftwaresuggestions.com/github/help-me-have-a-clean-message-when-replying-to-an-issue-via-email/

**Comment by zsloane** [comment]
Comments from email are just an absolute mess. Is there any way around this?

**Comment by icct-galvarez** [comment]
Hi, the "Reply to this email directly" used to work fine for me, but now (for the past few months), it includes the whole thread and renders the full text as HTML, including things like `<br>`, making it impossible to read:

<img width="757" alt="Screenshot 2025-03-06 at 11 38 51 AM" src="https://github.com/user-attachments/assets/5d916c7d-6729-43e8-89e7-d141a2bf579d" />

Is this possibly related to changes to Issues in https://github.com/orgs/community/discussions/148713?

**Comment by pta2002** [comment]
I want to add that this does not work at all if the email client is not in English. [Here](https://github.com/pta2002/typst-timeliney/issues/16#issuecomment-2784780141) is an instance of that happening to me. Since my email client (Android gmail in this case) is in Portuguese, it completely fails to parse the email history and it just gets included in the issue.

**Comment by jordenajames** [comment]
Tried replying to an issue via email for the first time and saw that it included the full "On [date], [email] wrote:" text in the thread. GitHub also stripped out the spacing, so my message got merged awkwardly—kind of like when a [food](https://sonicfoodmenus.com/) label smashes all the ingredients into one unreadable line. Plus, I couldn’t use markdown when editing on the web. Would be great if reply history was auto-trimmed and markdown was enabled for web edits.

**Comment by Oreeeee** [comment]
When I reply using e-mail from Blackberry Hub on Blackberry 10, it adds some trailing JSON data at the end for me.

**Comment by mrengy** [comment]
The reply by email functionality should also detect the [standard email signature separator](https://ladedu.com/how-to-use-the-standard-email-signature-separator/) “-- ” and truncate the text that comes after it. That would be an easy way to also exclude the entire email thread from the Github comment, where a signature with the standard separator is present.

**Comment by mailily031** [comment]
I had a similar experience—replying by email pulled in the full 'On [date], [email] wrote:' text, and the formatting got messy once GitHub stripped out the spacing. It really makes the thread harder to read. Having reply history auto-trimmed and markdown enabled for web edits would definitely improve the overall experience—almost like how labeling [delicious food](https://sonicmenu.us/) properly makes it so much easier to enjoy.
